reference/src/layout/unions.md

```diff
@@ -39,11 +39,56 @@ to obtain a pointer to any field, and vice versa.
 [6.5.8.5]: http://port70.net/~nsz/c/c11/n1570.html#6.5.8p5
 [6.7.2.1.16]: http://port70.net/~nsz/c/c11/n1570.html#6.7.2.1p16
 
-Since all fields are at offset 0, this implies that `repr(C)` unions do not have
-padding before or in-between their fields. They can, however, have trailing
-padding (see next example).
+#### Padding
```

Comment:
The precedent with `struct` I think is to use `**Paragraph Headers**`, not `#### Subsection headers`. I think that's better, the others become hard to distinguish when they get deeply nested.

Reply:
Actually looking at the rendered version, this looks fine.

What remains is a concern about consistency.

Reply:
I prefer to use `###`  for headings on markdown since those allow for easy linking, and semantically express a new heading or sub-section. If the rendered book makes those hard to distinguish, we could tweak its CSS, or report the issue to mdbook upstream. I'd rather open an issue to track this editorial change.

Reply:
Looking at https://rust-lang.github.io/unsafe-code-guidelines/layout/structs-and-tuples.html, `#### header` and `**Header**` actually have pretty much the same size.

So, fine for me. We should probably just adjust the other documents as well, eventually.